dnf/lock.py

```diff
@@ -64,6 +64,15 @@ def __init__(self, target, description, blocking=False):
         self.target = target
         self.thread_lock = threading.RLock()
 
+    def _try_ensure_dir(self):
```

Comment:
Or even better we can add ```makedirs``` to ```dnf.pycomp``` that has the ```exist_ok``` parameter and remove ```ensure_dir``` so we can just remove ```dnf.pycomp``` in the future and everything will be fine :)